File: lib/page/src/main/java/com/android/open9527/page/BindingVariableUtils.java
package com.android.open9527.page;

import android.util.SparseArray;

import androidx.databinding.ViewDataBinding;

/**
 * @author open_9527
 * Create at 2020/12/31
 **/
public final class BindingVariableUtils {


    public static void bindingVariable(ViewDataBinding binding, SparseArray bindingParams) {
        for (int i = 0, length = bindingParams.size(); i < length; i++) {
            binding.setVariable(bindingParams.keyAt(i), bindingParams.valueAt(i));
        }
    }
}

File: lib_kt/page/src/main/java/com/android/open9527/page/BindingVariableUtils.kt
package com.android.open9527.page

import android.util.SparseArray
import androidx.databinding.ViewDataBinding

/**
 *@author   open_9527
 *Create at 2021/5/18
 **/
object BindingVariableUtils {
    fun bindingVariable(binding: ViewDataBinding, bindingParams: SparseArray<*>) {
        var i = 0
        val length = bindingParams.size()
        while (i < length) {
            binding.setVariable(bindingParams.keyAt(i), bindingParams.valueAt(i))
            i++
        }
    }
}